File: Backend/productservice/src/main/java/sidharth/mishra/productservice/domain/Product.java
package sidharth.mishra.productservice.domain;

import lombok.Data;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.validation.constraints.NotBlank;
import java.util.Set;

@Document
@Data
public class Product {
    @Id
    @NotBlank
    private String productId;
    @NotBlank
    private String productName;
    @NotBlank
    private String cost;

    private Set<Images> productImages;
}

File: webapp/src/app/product.ts
import { Filehandle } from "./filehandle"

export interface Product {
    productId:string,
    productName:string,
    cost:string,
    productImages:Filehandle[]
    // constructor()
    // {
    //     this.productId="";
    //     this.productName="";
    //     this.cost="";
    // }
}
